github/github_test.go

```diff
@@ -42,9 +31,9 @@ const (
 // setup sets up a test HTTP server along with a github.Client that is
 // configured to talk to that test server. Tests should register handlers on
 // mux which provide mock responses for the API method being tested.
-func setup() {
-	// test server
-	mux = http.NewServeMux()
+func setup() (string, *http.ServeMux, *Client, func()) {
```

Comment:
Thanks, @chriskingnet! I was actually thinking about putting the comments into the `setup` func comments immediately following line 44, but that's OK.  As an alternative, I propose the following...

@shurcooL - for documentation purposes, what do you think about using named return values here?
I personally think it would make the code (and godocs) easier to understand:

```go
func setup() (url string, mux *http.ServeMux, client *Client, teardown func()) {
```

Reply:
Agreed, named returned values would help readability. Seeing `string` and `func()` doesn't say much.

Reply:
Thanks for the input on this, @shurcooL. Have now included this :)